distributedcombigrid/src/sgpp/distributedcombigrid/fullgrid/DistributedFullGrid.hpp

```diff
@@ -201,12 +201,103 @@ class DistributedFullGrid {
     MPI_Comm_free(&communicator_);
   }
 
+  FG_ELEMENT evalLocalIndexOn(const IndexVector& localIndex, const std::vector<real>& coords) const {
+    auto firstIndex = IndexVector(dim_, 0);
+    auto lastIndex = this->getLastGlobalIndex() - this->getFirstGlobalIndex();
+    // if this local index is out of bounds, return 0. (will be contributed by other partial dfg)
+    if (! (localIndex >= firstIndex && localIndex <= lastIndex) ) {
+      // std::cout << "out of bounds" << localIndex << firstIndex << lastIndex << std::endl;
+      return 0.;
+    }
+
+    // get coords corresponding to localIndex
+    auto localLinearIndex = getLocalLinearIndex(localIndex);
+    std::vector<real> pointCoords (this->getDimension());
+    getCoordsLocal(localLinearIndex, pointCoords);
+
+    // get product of 1D hat functions on coords
+    auto h = getGridSpacing();
+    real phi_c = 1.;
+    for (DimType d = 0 ; d < dim_ ; ++d){
+      // get distance between coords and point
+      pointCoords[d] -= coords[d];
+      if (std::abs(pointCoords[d]) > h[d]){
+        std::cout << "assert bounds " << pointCoords << coords <<
+         h << d << localIndex << lastIndex << std::endl;
+        assert(false &&
+          "should only be called for coordinates within the support of this point's basis function");
+      }
+      phi_c *= 1. - std::abs(pointCoords[d]/h[d]);
+    }
+    // std::cout << "coords " <<  localIndex << coords << localLinearIndex << h << std::endl;
+    // std::cout << "phi_c " << phi_c << this->getElementVector()[localLinearIndex] << std::endl;
+    assert(phi_c >= 0.);
+    return phi_c * this->getElementVector()[localLinearIndex];
+  }
+
+  FG_ELEMENT evalMultiindexRecursively (const IndexVector& localIndex, DimType dim, const std::vector<real>& coords) const {
```

Comment:
Maybe add discription on how to call function. It is not clear from the function that the index with lowest id in all dimensions should be used to start the recursion.

Reply:
hmm, actually you should rather call eval directly, which then calls the recursive function. maybe make this one private?